### PR TITLE
ens resolver

### PR DESCRIPTION
**Pull request by enriquesouza**
This resolves the ENS name to send

### CODE REVIEW

**Comment by enriquesouza** [comment]
Tests:
https://etherscan.io/name-lookup-search?id=vitalik.eth

Resolved Address:
 Address0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045 
 
 Test code:
 ```
 Task {
            let ensName = "vitalik.eth"
            let namehash = ensName.namehash()
            print("Namehash for \(ensName): \(namehash)")
            
            let factory = try EvmServiceFactory.getService(forChain: Chain.ethereum)
            let address = try await factory.resolveENS(ensName: ensName)
            
            print("Resolved address \(address)")
        }
 ```

File: VultisigApp/VultisigApp/Extensions/RpcEvmServiceEnsServiceExtension.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            
+            // Convert the address data to a hex string and return
+            return "0x" + addressData.toHexString()
+        } else {
+            throw RpcEvmServiceError.rpcError(code: -1, message: "Invalid address data")
+        }
+    }
+}
+
+extension String {
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Extensions/RpcEvmServiceEnsServiceExtension.swift:64]: we have a file call "StringExtensions.swift" why not put this method over there?

**Review by johnnyluo — APPROVED** [review]
This mostly looks OK to me , but I don't think this PR is complete , you still need to have logic on the UI part to actually resolve the input name to address.

I suppose you can raise another PR for that , I will approve this one for now

File: VultisigApp/VultisigApp/Extensions/RpcEvmServiceEnsServiceExtension.swift (hunk 1 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Extensions/RpcEvmServiceEnsServiceExtension.swift]: Moved

**Comment by enriquesouza** [comment]
> This mostly looks OK to me , but I don't think this PR is complete , you still need to have logic on the UI part to actually resolve the input name to address.
> 
> I suppose you can raise another PR for that , I will approve this one for now

I will break it down in back end and UI, easier to review. I will have to change in many places to validate the address.

**Pull request merged**